app/models/manageiq/providers/google/cloud_manager/vm.rb

```diff
@@ -1,6 +1,23 @@
 class ManageIQ::Providers::Google::CloudManager::Vm < ManageIQ::Providers::CloudManager::Vm
   include_concern 'Operations'
 
+  virtual_column :preemptible?, :type => :boolean, :uses => :advanced_settings
+
+  def preemptible?
+    preempt_setting = advanced_settings.detect { |v| v.name == "preemptible?" }
+
+    # It's possible that we got a nil back; this would occur if the VM hasn't
+    # been refreshed since this change has been added. If that's the case, log
+    # an error and return false for now.
+    if preempt_setting.nil?
+      _log.warn("Unable to find 'preemptible?' value for vm; this failure is"\
+                " likely temporary and will resolve upon a refresh")
+      return false
+    end
```

Comment:
I could make that change, but I like erring on the side of paranoia here with the warning message. If for some reason the code went south and always wrote nil, then at least we'd get warning messages about it (although honestly I bet they'd be lost in the noise).

I don't feel strongly either way though; if you'd prefer I can remove the log message.